packages/common/config/buildJestModuleMap.js

```diff
@@ -0,0 +1,6 @@
+module.exports = (existingModuleMap, library) => ({
+  ...existingModuleMap,
+  '\\.\\./ApolloProvider': library,
+  '\\.\\./ApolloConsumer': library,
+  '\\.\\./ApolloContext': library
```

Comment:
I'm not super happy with this approach, but I haven't found a better workaround (without sacrificing in other areas). Here we're basically overriding relative calls to source included within a package, to hijack the resolution of the source to instead pull from either the CJS or UMD bundle. While this works, it means that anytime we want to reference a file using a relative path in a test file, we have to make sure we've added a matching line in the `buildJestModuleMap.js` file to properly CJS/UMD test it. Right now this isn't that big of a deal since the number of files we have to hijack is pretty small, and isn't likely going to change anytime soon. But ideally we'd have a more generic way to intercept and redirect the lookup for all relative files under test.

We could consider always including `src` or some other common prefix in imports, then intercept based on `src` being in the lookup reference (like current day React Apollo [is doing](https://github.com/apollographql/react-apollo/blob/0adeb79e886003f57d0f77df875923cc75b4c332/jest.cjs.config.js#L3-L9)). Due to the fact that RA 3 leverages `__tests__` directories within `src` however, this would mean imports like

```js
import { graphql } from '../../src/graphql';
```

which just look strange, given that 

```js
import { graphql } from '../graphql';
```

is what most people would expect to see/use. I'm definitely open to better suggestions here.

Reply:
We're going with a different (more dynamic) approach for this, in https://github.com/apollographql/react-apollo/pull/2892/commits/6cf9f28409edf72d14a47fd5ab2865ee00c45809.